js/date-picker/js/lumx.date_picker_directive.js

```diff
@@ -28,6 +28,17 @@ angular.module('lumx.date-picker', [])
             $scope.days = [];
             $scope.daysOfWeek = [$scope.localeData._weekdaysMin[1], $scope.localeData._weekdaysMin[2], $scope.localeData._weekdaysMin[3], $scope.localeData._weekdaysMin[4], $scope.localeData._weekdaysMin[5], $scope.localeData._weekdaysMin[6], $scope.localeData._weekdaysMin[0]];
 
+            // to display year selection, taken range -100 to +100 of current year
+            var currentYear = moment().locale(locale).year();
+            // store current selected year(it will be used while selecting year)
+            $scope.currentSelectedYear = currentYear;
+            var startingYear = currentYear - 100;
+            var endingYear = currentYear + 100;
+            $scope.years = [];// it will contain years to be displayed
+            for(var c=startingYear;c<=endingYear;c++){
```

Comment:
Bad spacing. It's not Allman style and we prefer to have a space after the `for`, around the `=` and after the `;`.